extensions/config-yaml/runtime/src/main/java/io/quarkus/yaml/configuration/runtime/YamlConfigSource.java

```diff
@@ -0,0 +1,127 @@
+package io.quarkus.yaml.configuration.runtime;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Serializable;
+import java.io.UncheckedIOException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.StringJoiner;
+import java.util.TreeMap;
+
+import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.yaml.snakeyaml.Yaml;
+
+public class YamlConfigSource implements ConfigSource, Serializable {
+
+    private static final long serialVersionUID = -875865410658150582L;
+
+    private static final String CONFIG_ORDINAL_KEY = "config_ordinal";
+    private static final String CONFIG_ORDINAL_DEFAULT_VALUE = "100";
+    private static final String PROPERTY_KEY_SEPARATOR = ".";
+    private static final String COMMA = ",";
+    private static final String ESCAPED_COMMA = "\\,";
+
+    private final URL source;
+    private final int ordinal;
+    private final Map<String, String> properties;
+
+    public YamlConfigSource(URL source, int ordinal) {
+        this.source = source;
+        this.properties = buildConfigPropertiesFromYaml();
+        if (properties.containsKey(CONFIG_ORDINAL_KEY)) {
+            this.ordinal = Integer.parseInt(properties.getOrDefault(CONFIG_ORDINAL_KEY, CONFIG_ORDINAL_DEFAULT_VALUE));
+        } else {
+            this.ordinal = ordinal;
+        }
+    }
+
+    private Map<String, String> buildConfigPropertiesFromYaml() {
+        if (source == null) {
+            return Collections.emptyMap();
+        }
+        try (InputStream is = source.openStream()) {
+            @SuppressWarnings("unchecked")
+            Map<String, Object> yamlContent = new Yaml().loadAs(is, HashMap.class);
```

Comment:
```suggestion
            Map<String, Object> yamlContent = new Yaml().loadAs(is, TreeMap.class);
```

Minor lookup optimization